articles/azure/azs-how-create-vm-image-powershell.md

```diff
@@ -0,0 +1,171 @@
+$ImageName "<output form="imagename" name="result" style="display: inline;">MyCustomImage</output>"
+
+# Get VM details
+$VM = Get-AzureRmVM -Name $VMName -ResourceGroupName $RGName
+
+if ($VM) {
+    # Stop the VM
+    Write-Output -InputObject "Stopping VM and marking as generalised..."
+    Stop-AzureRmVM -Name $VMName -ResourceGroupName $RGName
+
+    # Mark VM as Generalised
+    Set-AzureRmVM -Name $VMName -ResourceGroupName $RGName -Generalized
+}
+else {
+    Write-Error -Message "VM with name $VMName does not exist in resource group $RGName"
+    break
+}
+
+# Get VM details again
+$VM = Get-AzureRmVM -Name $VMName -ResourceGroupName $RGName
+
+# Create image
+Write-Output -InputObject "Creating image of VM: $VMName"
+$ImageConfig = New-AzureRmImageConfig -Location $Location -SourceVirtualMachineId $VM.Id
+$Image = New-AzureRmImage -ResourceGroupName $RGName -ImageName $ImageName -Image $ImageConfig
+
+# Declare variables
+$NewVMName = "<output form="newvmname" name="result" style="display: inline;">MyNewVMFromImage</output>"
+$VNetName = "<output form="vnetname" name="result" style="display: inline;">MyVNetwork</output>"
+$SubnetName = "<output form="subnetname" name="result" style="display: inline;">MySubnet</output>"
+$NSGName = "<output form="nsgname" name="result" style="display: inline;">MyNSG</output>"
+$PublicIPName = "<output form="publicipname" name="result" style="display: inline;">MyPublicIP</output>"
+$Username = "<output form="vmusername" name="result" style="display: inline;">MyUser</output>"
+$Password = "<output form="vmpassword" name="result" style="display: inline;">Password123!</output>" | ConvertTo-SecureString -Force -AsPlainText
+$Credential = New-Object PSCredential($Username, $Password)
+
+# Get image to check OS type.
+$Image = Get-AzureRmImage | Where-Object { $_.Name -like $ImageName }
+
+# If Windows, open RDP port and provision correct size 
+if ($Image.StorageProfile.OsDisk.OsType -like "Windows") {
+    $OpenPorts = 3389
+    if (-not $Size) {
+        $Size = "Standard_DS2_v2"
+    }
+}
+# OsType is Linux and open SSH port and provision correct size
+else {
+    $OpenPorts = 22
+    if (-not $Size) {
+        $Size = "Standard_DS1_v2"
+    }
+}
+
+# Create new VM from custom image
+Write-Output -InputObject "Creating VM from image: $ImageName... (This may take a while)"
+New-AzureRmVM -ResourceGroupName $RGName -Location $Location -Name $NewVMName -ImageName $ImageName -Credential $Credential -VirtualNetworkName $VNetName -SubnetName $SubnetName -PublicIpAddressName $PublicIPName -SecurityGroupName $NsgName -OpenPorts $OpenPorts -Size $Size
+</code></pre>
+
+***
+
+### [Windows](#tab/tabid-a/tabid-1)
+
+1. Log in to your Windows VM.
+
+2. Run the following command: `C:\Windows\System32\Sysprep\sysprep.exe`.
+
+3. In the **System Preparation Tool**, under *System Cleanup Action*, select **Enter System Out-of-Box Experience (OOBE)** from the dropdown menu.
```

Comment:
In the **System Preparation Tool**, under *System Cleanup Action*, select **Enter System Out-of-Box Experience (OOBE)** from the **System Cleanup Action** list.